src/PolyType.SourceGenerator/PolyTypeGenerator.cs

```diff
@@ -21,13 +22,30 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         IncrementalValueProvider<PolyTypeKnownSymbols> knownSymbols = context.CompilationProvider
             .Select((compilation, _) => new PolyTypeKnownSymbols(compilation));
 
+        IncrementalValuesProvider<TypeExtensionModel> externalTypeShapeExtensions = context.MetadataReferencesProvider
+            .Combine(context.CompilationProvider)
+            .Combine(knownSymbols)
+            .SelectMany((tuple, token) => Parser.DiscoverTypeShapeExtensions(tuple.Right, tuple.Left.Right, tuple.Left.Left, token));
+
+        IncrementalValuesProvider<TypeExtensionModel> ownTypeShapeExtensions = context.CompilationProvider
```

Comment:
Can these two incremental values be fused into just one? It seems they both depend on MetadataReferencesProvider and CompilationProvider.

Reply:
I'm guessing it has something to do with the logic starting in line 34. Assuming there is a way we could fuse some of this logic so that the source generator defines a smaller incremental graph overall that would be great.

Reply:
The more I look at this the more I think it can be simplified. The `Compilation` objects already encapsulates all metadata references via the [Compilation.References property](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.references?view=roslyn-dotnet-4.9.0) so to me it seems all this logic can be removed from the incremental graph and hidden inside the `Parser` logic.

Reply:
Sounds good. I added this before I understood that the Parser was doing everything on every compilation anyway.